src/main/java/software/amazon/smithy/gradle/SmithyPlugin.java

```diff
@@ -39,49 +40,35 @@
     private static final String DEFAULT_CLI_VERSION = "0.9.4";
     private static final List<String> SOURCE_DIRS = ListUtils.of(
             "model", "src/$name/smithy", "src/$name/resources/META-INF/smithy");
-
-    private boolean appliedPlugin;
+    private static final Logger LOGGER = Logger.getLogger(SmithyPlugin.class.getName());
 
     @Override
     public void apply(Project project) {
+        // Ensure that the Java plugin is applied.
         project.getPluginManager().apply(JavaPlugin.class);
-        project.getPluginManager().withPlugin("java", javaPlugin -> {
```

Comment:
This change ensures that the "smithy" extension and "smithyBuildJar" task are registered right away, allowing them to be configured and disabled in build.gradle files.